src/api/control/mod.rs

```diff
@@ -2,6 +2,70 @@
 //!
 //! [Control API]: http://tiny.cc/380uaz
 
-mod member;
+pub mod endpoint;
+pub mod member;
+pub mod pipeline;
+pub mod room;
 
-pub use self::member::{Id as MemberId, Member};
+use std::{convert::TryFrom as _, fs::File, io::Read as _, path::Path};
+
+use failure::{Error, Fail};
+use serde::Deserialize;
+
+use self::pipeline::Pipeline;
+
+pub use self::{
+    endpoint::Endpoint,
+    member::{Id as MemberId, MemberSpec},
+    room::{Id as RoomId, RoomElement, RoomSpec},
+};
+use std::fs::ReadDir;
+
+#[derive(Clone, Deserialize, Debug)]
+#[serde(tag = "kind")]
+pub enum RootElement {
+    /// Represent [`RoomSpec`].
+    /// Can transform into [`RoomSpec`] by `RoomSpec::try_from`.
+    Room {
+        id: RoomId,
+        spec: Pipeline<RoomElement>,
+    },
+}
+
+/// Errors that can occur when we try transform some spec from `Element`.
+/// This error used in all [`TryFrom`] of Control API.
+///
+/// [`TryFrom`]: std::convert::TryFrom
+#[allow(clippy::pub_enum_variant_names)]
+#[derive(Debug, Fail)]
+pub enum TryFromElementError {
+    #[fail(display = "Element is not Room")]
+    NotRoom,
+    #[fail(display = "Element is not Member")]
```

Comment:
Use `#[display(fmt = ...)]` from `derive_more` instead of `#[fail(display = ...)]`.